File: test/pratica04/app/src/main/java/com/example/pratica4/MainActivity.java
package com.example.pratica4;

import androidx.appcompat.app.AppCompatActivity;

import android.os.Bundle;
import android.view.View;
import android.widget.Button;
import android.widget.EditText;
import android.widget.Toast;

public class MainActivity extends AppCompatActivity {

    private EditText txtUsuario;
    private EditText txtSenha;
    private Button btnEntrar;
    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main);
        txtUsuario = (EditText) findViewById(R.id.txtUsuario);
        txtSenha = (EditText) findViewById(R.id.txtSenha);
        btnEntrar = (Button) findViewById(R.id.btnEntrar);

        btnEntrar.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View v) {
                String usuario = txtUsuario.getText().toString();
                String senha = txtSenha.getText().toString();

                if (!usuario.equals("") && !senha.equals("")) {
                    if(usuario.equals("admin") && senha.equals("1234")) {
                        Toast.makeText( MainActivity.this,  "OK", Toast.LENGTH_LONG).show();
                    }
                }else{
                    Toast.makeText( MainActivity.this,  "Preencha todos os campos", Toast.LENGTH_LONG).show();
                }


            }
        });

    }
}

File: src/android/aula31_32/formularios/app/src/main/java/br/iesb/ads023/formularios/MainActivity.kt
package br.iesb.ads023.formularios

import androidx.appcompat.app.AppCompatActivity
import android.os.Bundle
import android.widget.Button
import android.widget.EditText

class MainActivity : AppCompatActivity() {
    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_main)
        setTitle("Formulários")

        var btnEntrar = findViewById<Button>(R.id.btnEntrar)
        btnEntrar.setOnClickListener {
            if (validarFormulario())
                println("Sucesso!")
        }
    }

    fun validarFormulario(): Boolean {
        var valido = true
        val edtMatricula = findViewById<EditText>(R.id.edtMatricula)
        val edtSenha = findViewById<EditText>(R.id.edtSenha)

        if (edtMatricula.text.toString().length == 0) {
            edtMatricula.error = "Matrícula obrigatória"
            valido = false
        }

        when {
            edtSenha.text.toString().length == 0 -> {
                edtSenha.error = "Senha obrigatória"
                valido = false
            }
            edtSenha.text.toString().length < 6 -> {
                edtSenha.error = "Senha menor que 6 caracteres"
                valido = false
            }
        }

        return valido
    }
}